services/wallet/activity/fetched_entries.go

```diff
@@ -75,6 +76,22 @@ func getFetchedEntriesByIDs(ctx context.Context, deps FilterDependencies, txIDs
 		return nil, fmt.Errorf("failed to convert rows to transfers: %w", err)
 	}
 
+	// Request async token discovery for ERC20 transfers to make sure we have their metadata
+	if deps.tokenPublisher != nil {
```

Comment:
hmmm, the way we normally do it, modules publish events in their "domain" (alchemy knows deals with transfers, so it publishes events like "hey I've just fetched these transfers". 

It is the subscriber's responsibility to interpret those events and react in accordance (that is, tokenManager should subscribe to "here's some new transfers" signal and know it needs to potentially fetch metadata for some new token). The benefit of this becomes clearer when there's multiple subscribers for a given publisher, it doesn't make sense for the publisher to have knowledge of every subscriber.

Reply:
also, requesting the token metadata every time the client fetches activity entries seems like a bit of an overkill. Can't we detect tokens for which for some reason the metadata is not available and only request those? You can do that request directly (ie call some method from an interface) instead of using the pubsub mechanism for that.